src/huggingface_hub/_commit_api.py

```diff
@@ -246,6 +251,29 @@ def b64content(self) -> bytes:
         with self.as_file() as file:
             return base64.b64encode(file.read())
 
+    @property
+    def _local_oid(self) -> Optional[str]:
```

Comment:
out of curiosity, how long does this take to compute on larger files?

Reply:
Asking as large file uploads (without going through your fantastic large-file-upload method) are already taking quite a bit of time before uploading anything

Reply:
For LFS files (so any file larger than 10MB for text files or 1MB for binary files) it won't add any time as we already compute and have the sha256.
For non-LFS files i expect this to be quite fast unless there are many many small files

Reply:
As @julien-c said, it should be neglectable on regular files since they are small. In any case, all regular files are loaded in memory in the `/commit` payload and cannot be above 1GB so that's not a use case to optimize for.